recipes/fftw/all/conanfile.py

```diff
@@ -70,64 +87,93 @@ def generate(self):
         tc.variables["ENABLE_OPENMP"] = self.options.openmp
         tc.variables["ENABLE_THREADS"] = self.options.threads
         tc.variables["WITH_COMBINED_THREADS"] = self.options.get_safe("combinedthreads", False)
-        tc.variables["ENABLE_FLOAT"] = self.options.precision == "single"
-        tc.variables["ENABLE_LONG_DOUBLE"] = self.options.precision == "longdouble"
         tc.variables["ENABLE_SSE"] = self.options.simd == "sse"
         tc.variables["ENABLE_SSE2"] = self.options.simd == "sse2"
         tc.variables["ENABLE_AVX"] = self.options.simd == "avx"
         tc.variables["ENABLE_AVX2"] = self.options.simd == "avx2"
         tc.generate()
 
+    @property
+    def build_folder(self):
+        bf = super().build_folder
+        return os.path.join(bf, self._current_precision) if self._current_precision else bf
+
+    @property
+    def _all_precisions(self):
+        return [p for p in ALL if self.options.precision == p or self.options.get_safe(f"precision_{p}")]
```

Comment:
```suggestion
        return [p for p in ALL if self.options.get_safe(f"precision_{p}")]
```

Reply:
that will not produce the same effect - specifying `precision=deprecated` will produce package with no libraries at all (=useless package).